### PR TITLE
ENH: Implement convert_dtypes on block level

### PR DESCRIPTION
**Pull request by phofl**
The current implementation is pretty bad in the sense that we will always end up splitting blocks even if we don't want to touch the block at all. Pushing this to the Block level let's us solve that problem.

### CODE REVIEW

**Comment by phofl** [comment]
cc @mroeschke

File: pandas/core/internals/blocks.py [hunk 1]
```diff
@@ -665,12 +714,18 @@ def astype(
...
+            squeeze values to ndim=1 if only one column is given
 
         Returns
         -------
         Block
         """
         values = self.values
+        if squeeze and values.ndim == 2:
+            if values.shape[0] != 1:
+                raise ValueError("Can not squeeze with more than one column.")
```
Comment (mroeschke) [inline pandas/core/internals/blocks.py:727]: Is this just a defensive exception i.e. we should never hit this?

Comment (phofl) [reply pandas/core/internals/blocks.py:727]: yes, we shouldn't get there

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**